zero-downtime-schema-migrations-postgresql.mdx

```diff
@@ -0,0 +1,119 @@
+
+**Planned downtime**
+
+Then, there’s always the possibility that downtime is acceptable for your application and business. Maybe your audience is only around during business hours or you have a regular planned maintenance every month. In which case, you can take more of a “big bang” approach. Multi-step process or intimately integrated ORMs simply aren’t necessary to avoid downtime, giving you more time to get creative with your maintenance window page.
+
+## Tools available for schema migrations
+
+When you start seriously considering more destructive changes to your schema, thrid party tooling becomes more of a necessity to keep track of the changes, version them and perform the migrations. Since this problem has been around for awhile, there are a variety of tools. We won't go into details for each one, but some solutions have been around for nearly two decades and others popped up in the last few years.
+
+![Tools for schema migrations](/images/zdt-schema-tools.png)
+
+Both [Liquibase](https://liquibase.com/) and [Flyway](https://flywaydb.org/) have been in this space for some time, Liquibase having been here the longest since 2006. Both of these offerings have a free open source solution. That being said, there are numerous features ranging from rollbacks to schema diffs that are gated at the paid tier. Liquibase is more geared towards enterprise use cases, and Flyway being a bit more developer oriented. Both solutions can use SQL to write migration scripts (though Liquibase supports other formats – YAML, XML, JSON.) and support a wide variety of databases.
+
+[Sqitch](https://sqitch.org/) is interesting because it’s positioned itself as a standalone change management system. It is unopinionated about your application framework, database engine or development environment. Unlike liquibase or flyway, there is no need to number or name your migrations a certain way. Sqitch handles this for you. It’s a unique solution that is aims to make it easy to integrate into your existing environment
+
+[Atlas](https://atlasgo.io/) and [Bytebase](https://bytebase.com/) are very much the new kids on the block and lean into developer experience and database as code. Both have open source and cloud offerings and aim to do much more than schema migrations, supporting multiple databases.
+
+Last but not least is [Reshape](https://github.com/fabianlindfors/reshape). This is a relatively new open source tool that provides a novel way for handling schema migrations. Though unlike these other tools, it doesn’t support multiple databases. The tool is only for Postgres
+This means it can take advantage of capabilities specific to Postgres moreso than other solutions. Reshape inspired the next tools we'll dive into, pgroll.
```

Comment:
```suggestion
This means it can take advantage of capabilities specific to Postgres moreso than other solutions. Reshape inspired the next tools we'll dive into: pgroll.
```